index.js

```diff
@@ -135,6 +136,41 @@ class Inspector {
       this.dhtServer = null
     }
   }
+
+  async writeHeapSnapshot (fileLocation = null) {
+    // Based on https://nodejs.org/api/inspector.html#heap-profiler
+    if (!fileLocation) {
+      fileLocation = `${this.filename}-${Date.now()}.heapsnapshot`
+    }
+
+    const session = new this.inspector.Session()
+    const file = openSync(fileLocation, 'w')
+    try {
+      session.connect()
```

Comment:
Is there a reason this one is inside a try-catch?

If it was not, then we could remove the outer try-catch.

Reply:
The outer try-finally ensures the file is always closed, the inner try-finally ensures the session is always destroyed

Reply:
But does the outer `try` need to be there? Couldn't it just be `closeSync()` after the inner `try`.

Reply:
Ah, I see what you mean now.  The outer try protects against the case where `session.connect` throws.
With a single try-finally containing everything, the finally would need to figure out whether the `connect()` call succeeded (to know whether it should call `session.destroy()`), so using the double finally felt easiest